### PR TITLE
Add wiremock module

### PR DESCRIPTION
**Pull request by mcruzdev**
Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [our code style](CONTRIBUTING.md#code-style)
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] Subject`
- [x] Pull Request contains link to the issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-quarkus2` to backport the original PR to the `quarkus2` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by mcruzdev** [comment]
Hi @hbelmiro and @ricardozanini I created the wiremock module, what initial features we need to have to merge to main?

Is it possible to create a branch for wiremock stuff, or just adding an initial feature is ok for now?

**Comment by hbelmiro** [comment]
@mcruzdev 

> what initial features we need to have to merge to main?

From us, you're probably the one who knows more about it. I'll let you decide.
Is this wiremock support a real need for you as a user? If so, what are the ones you need the most? From those, pick the easiest/fastest to implement.

> Is it possible to create a branch for wiremock stuff, or just adding an initial feature is ok for now?
IMO, we can add features incrementally (an initial feature is ok), so users can benefit from them ASAP.

**Comment by mcruzdev** [comment]
Hi @hbelmiro, Thank you!

It help me a lot, I will continue here!

**Comment by mcruzdev** [comment]
- [ ] Add response status from OpenAPI
- [ ] Add response body from OpenAPI (without to get from components)

**Comment by mcruzdev** [comment]
Hi @hbelmiro and @ricardozanini, I think that we can get a first review about this new feature.

**Comment by ricardozanini** [comment]
I'd rather wait for #482 to be merged first. @carlesarnal will kick my ass if I keep asking him to rebase.

File: wiremock/deployment/pom.xml
Comment (mcruzdev) [reply wiremock/deployment/pom.xml:17]: This is the same used by `client`.

**Comment by mcruzdev** [comment]
> Just a few comments initially, I'll do a better review later.

I switched the extension to `experimental` Sound good?

I've added some basic features for starters. It is great for new contributors!

File: wiremock/deployment/pom.xml
Comment (carlesarnal) [reply wiremock/deployment/pom.xml:17]: No, we can't replace this.

**Comment by ricardozanini** [comment]
@mcruzdev can you rebase? 🙏

**Comment by ricardozanini** [comment]
@mcruzdev are the issues pointed by GH relevant?

**Comment by mcruzdev** [comment]
> @mcruzdev are the issues pointed by GH relevant?

I think we can ignore because are Test classes. makes sense?

> Unused class: SchemaReaderTest is not referenced within this codebase. If not used as an external API it should be removed.

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Request.java
Comment (hbelmiro) [inline wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Request.java:5]: Can't this be a `record`?

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Response.java
Comment (hbelmiro) [inline wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Response.java:6]: Can't this be a `record`?

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Stubbing.java
Comment (hbelmiro) [inline wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Stubbing.java:10]: I think we should rename it to `Stub`.

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Stubbing.java
Comment (hbelmiro) [inline wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Stubbing.java:25]: Do we really need a builder?

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wrapper/OpenApiWiremockGeneratorWrapper.java
Comment (hbelmiro) [inline wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wrapper/OpenApiWiremockGeneratorWrapper.java:29]:
```suggestion
    private static final String WIREMOCK_OUTPUT_FILE = "wiremock-stubs.json";
```

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wrapper/OpenApiWiremockGeneratorWrapper.java
Comment (hbelmiro) [inline wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wrapper/OpenApiWiremockGeneratorWrapper.java:77]: Are you sure we don't need to abort the entire generation here?

File: wiremock/docs/antora.yaml
Comment (hbelmiro) [inline wiremock/docs/antora.yaml:2]:
```suggestion
title: OpenAPI Generator
```

File: wiremock/docs/modules/ROOT/nav.adoc
Comment (hbelmiro) [inline wiremock/docs/modules/ROOT/nav.adoc:1]:
```suggestion
* xref:index.adoc[Quarkus - OpenAPI Generator - Wiremock]
```

File: wiremock/docs/modules/ROOT/pages/index.adoc
Comment (hbelmiro) [inline wiremock/docs/modules/ROOT/pages/index.adoc:1]:
```suggestion
= Quarkus - OpenAPI Generator - Wiremock
```

File: wiremock/integration-tests/pom.xml
Comment (hbelmiro) [inline wiremock/integration-tests/pom.xml:13]:
```suggestion
    <name>Quarkus - OpenAPI Generator - Wiremock - Integration Tests</name>
```

File: wiremock/pom.xml
Comment (hbelmiro) [inline wiremock/pom.xml:13]:
```suggestion
    <name>Quarkus - OpenAPI Generator - Wiremock - Parent</name>
```

File: wiremock/runtime/pom.xml
Comment (hbelmiro) [inline wiremock/runtime/pom.xml:12]:
```suggestion
    <name>Quarkus - OpenAPI Generator - Wiremock</name>
```

**Review by hbelmiro — COMMENTED** [review]
Good job and amazing contribution @mcruzdev. Thank you!
I left some comments.

**Comment by mcruzdev** [comment]
Hi @hbelmiro thank you for the reviewing it, I will take a look soon!

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Response.java
Comment (mcruzdev) [reply wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Response.java:6]: Hi @hbelmiro, yeah can be. Another point here is the use of `@JsonInclude` annotation, I do no know if makes sense to use it, I will try to remove it and use another way to not include non empty.

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/OpenApi2WiremockMapper.java
Comment (mcruzdev) [reply wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/OpenApi2WiremockMapper.java:81]: No Helber, I wanted to follow the OpenAPI specification name, I can change here.

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wrapper/OpenApiWiremockGeneratorWrapper.java
Comment (mcruzdev) [reply wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wrapper/OpenApiWiremockGeneratorWrapper.java:77]: We are generating one Wiremock stub for one OpenAPI specification file, the generation is independent, if one fails does not affect the another one, in your experience, is better make the generation dependent?

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/OpenApi2WiremockMapper.java
Comment (hbelmiro) [reply wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/OpenApi2WiremockMapper.java:81]: The compiler uses `$` to name inner classes. I'm not sure it would cause issues in real life, but I'd avoid it.
It may also confuse log readers as well. They may think that `Ref` is an inner class.

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wrapper/OpenApiWiremockGeneratorWrapper.java
Comment (hbelmiro) [reply wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wrapper/OpenApiWiremockGeneratorWrapper.java:77]: In general, I prefer failing "atomically".
In this particular case, it doesn't seem like users would need to generate just a few stubs. I think they'd like to generate them all. (I would, as a user. But I may be wrong.)
Also, if we just log the exception, users may not notice it occurred.
I don't have a strong opinion here, but I would generate them all or fail (not just log the error).

**Comment by hbelmiro** [comment]
@mcruzdev can you please let me know when I can re-review it?

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/OpenApi2WiremockMapper.java
Comment (mcruzdev) [reply wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/OpenApi2WiremockMapper.java:81]: It makes total sense!

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/SchemaReader.java
Comment (mcruzdev) [reply wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/SchemaReader.java:15]: I changed this class to be an interface instead

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/SchemaReader.java
Comment (mcruzdev) [reply wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/SchemaReader.java:20]: I did not understand this point @hbelmiro, can you explain in more details?

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Stubbing.java
Comment (mcruzdev) [reply wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Stubbing.java:25]: Actually no, we can add it when we have more attribute here.

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wrapper/OpenApiWiremockGeneratorWrapper.java
Comment (mcruzdev) [reply wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wrapper/OpenApiWiremockGeneratorWrapper.java:77]: It help me to think, makes sense!

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/OpenApi2WiremockMapper.java
Comment (hbelmiro) [inline wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/OpenApi2WiremockMapper.java:3]: Please don't use star imports.

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/SchemaReader.java
Comment (hbelmiro) [reply wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/SchemaReader.java:20]: We can remove the `Map` parameter of `mapObjectExample`.
By doing this, we avoid modifying objects passed as an argument, increasing the code's immutability.
Like this:

```java
    static String readObjectExample(Schema<?> schema) {
        try {
            Map<String, Object> map = mapObjectExample(schema);
            return OpenApiWiremockGeneratorWrapper.OBJECT_MAPPER_INSTANCE.writeValueAsString(map);
        } catch (JsonProcessingException e) {
            return EMPTY_JSON_OBJECT;
        }
    }

    private static Map<String, Object> mapObjectExample(Schema<?> example) {
        HashMap<String, Object> root = new HashMap<>();
        Optional.ofNullable(example.getProperties()).orElse(Map.of())
                .forEach((key, schema) -> {
                    if (schema.getType().equals(OBJECT_TYPE)) {
                        root.put(key, mapObjectExample(schema));
                    } else {
                        root.put(key, schema.getExample());
                    }
                });
        return root;
    }
```

File: wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Request.java
Comment (hbelmiro) [inline wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Request.java:12]: It seems like this can be removed.

Comment (mcruzdev) [reply wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wiremock/model/Request.java:12]: >We can remove the Map parameter of mapObjectExample.
By doing this, we avoid modifying objects passed as an argument, increasing the code's immutability.
Like this:

It makes total sense, TY for this.

**Review by hbelmiro — APPROVED** [review]

**Comment by mcruzdev** [comment]
Hi @hbelmiro and @ricardozanini I think we can wait for merge this pull request, I am seeing here some projects that does similar things and I am collecting some ideas to use here... Another point is: I am very busy and I think if we merge it we will not have effort to upgrade this module faster. WDYT?

**Pull request closed without merging**
